validmind/ai.py

```diff
@@ -0,0 +1,204 @@
+# Copyright © 2023-2024 ValidMind Inc. All rights reserved.
+# See the LICENSE file in the root of this repository for details.
+# SPDX-License-Identifier: AGPL-3.0 AND ValidMind Commercial
+
+import concurrent.futures
+import os
+
+from openai import AzureOpenAI, OpenAI
+
+from .utils import clean_docstring
+
+SYSTEM_PROMPT = """
+You are an expert data scientist and MRM specialist tasked with providing concise and'
+objective insights based on the results of quantitative model or dataset analysis.
+
+Examine the provided statistical test results and compose a brief summary. Highlight crucial
+insights, focusing on the distribution characteristics, central tendencies (such as mean or median),
+and the variability (including standard deviation and range) of the metrics. Evaluate how
+these statistics might influence the development and performance of a predictive model. Identify
+and explain any discernible trends or anomalies in the test results.
+
+Your analysis will act as the description of the result in the model documentation.
+
+Avoid long sentences and complex vocabulary.
+Structure the response clearly and logically.
+Use Markdown syntax to format the response.
+Use the Test ID that is provided to form the Test Name e.g. "ClassImbalance" -> "Class Imbalance".
+Use the following format for the response:
+```
+**<Test Name>** is a **<Test Type>** that <continue to explain what it does in detail>...
```

Comment:
```suggestion
**<Test Name>** <continue to explain what it does in detail>...
```

I feel like test type is probably not needed in this context. The name of the test should suffice.

Reply:
yep that makes sense